velox/exec/Task.h

```diff
@@ -37,6 +37,16 @@ using ConnectorSplitPreloadFunc =
 
 class Task : public std::enable_shared_from_this<Task> {
  public:
+  /// Threading mode the task is executed.
+  enum class ExecutionMode {
+    /// Mode that executes the query serially (single-threaded) on the calling
+    /// thread.
+    kSerial,
+    /// Mode that executes the query in parallel (multi-threaded) using provided
+    /// executor.
```

Comment:
nit: add: Task is executed via the Task::start() API that starts up multiple driver threads and manages their lifecycle.